### ISSUE TITLE
Q: about download URLS security

### ISSUE THREAD

**Issue by increpare** [itchio/itch.io#373]
Hello! I was gonna try put a new game I'm working on there for sale, but I don't understand how secure the download urls are.  For a high-cost game ($30), there'll possibly be a some incentive for people to share links - are there any measures to prevent against abuse?  I couldn't find an answer on the site.

I understand itch is generally for smaller scale things, where it does amazingly - I don't expect it to cater to bigger things, but I thought I'd ask how it handles things right now.

<3

S

**Comment by leafo** [comment]
Every purchase gets a page with a unique URL. If you purchase something with an account on itch.io then that page is associated to your account and can only be loaded when logged in. By default there are no restrictions on downloads to those pages.

In practice we've seen very little to no abuse on the purchase pages. If we required everyone to have accounts any sharing issues would still remain since owners could just share a common account. We also don't enforce any DRM, so if a malicious buyer wanted to distribute your game they could download and attempt to download once and host elsewhere. Having us disable a download page might deter some cases, but the reality is that if someone wants to  do an unauthorized distribution, they will figure out something. Adding restrictions to the pages on itch.io might more often frustrate legitimate buyers so we've haven't done anything by default.

From your purchase page you can see how many times people have downloaded each purchase. If you notice any suspicious behavior we can investigate and/or disable the key temporarily or permanently.

The direct links to the files themselves can not be shared, they expired after a few seconds.

If you're interested in having default restrictions on your purchase pages I wouldn't be against building some optional tools that developers can enable.

**Comment by increpare** [comment]
Firstly, thanks for the clear +prompt reply!

A way to quickly view + sort keys by # downloads might save some legwork
(from what I can tell, right now you can't see the number of downloads
except by clicking on each purchase manually)?

Best,

S

**Comment by leafo** [comment]
That's correct, you have to click into the purchase to see the download counts. Is your plan to monitor your purchases listing looking for suspicious behavior?

**Comment by increpare** [comment]
>Is your plan to monitor your purchases listing looking for suspicious
behavior?
If I put it on itch then yeah, as per your suggestion, I would feel like I
should check up every so often.

**Comment by leafo** [comment]
Ideally that's something that's automated, not something you should feel the need to check up on. Would you prefer to be some built in limit?

**Comment by increpare** [comment]
Apologies for the delayed reply. I don't know. I like itch the way it is,
and don't want to petition for a potential culture change, even if that
happens to make it less suitable for my particular use case. :|

**Comment by Guard13007** [comment]
How about instead of a built-in *limit*, a level where the developer will be notified? Something like x downloads in the past x days triggers a warning in the dashboard.

**Comment by increpare** [comment]
Yip, that might be ok (better still if there were email notifications).

What if one cancels someone's access and they complain?
